File: Feitos em Java/1091.java
import java.io.IOException;
import java.util.Scanner;
public class Main {
    public static void main(String[] args) throws IOException {
        Scanner leitor = new Scanner(System.in);
        while(true){
            int quant = leitor.nextInt();
            if(quant==0){
                break;
            }
            int dX = leitor.nextInt();
            int dY = leitor.nextInt();
            for(int i=0;i<quant;i++){
                int x = leitor.nextInt();
                int y = leitor.nextInt();
                if(x==dX||y==dY){
                    System.out.println("divisa");
                }
                else if(x>dX&&y>dY){
                    System.out.println("NE");
                }
                else if(x<dX&&y>dY){
                    System.out.println("NO");
                }
                else if(x<dX&&y<dY){
                    System.out.println("SO");
                }
                else if(x>dX&&y<dY){
                    System.out.println("SE");
                }
            }
	}
    }
}

File: Feitos em C/1091.c
#include <stdio.h>
int main() {
	int quant,i;
	while(2>1){
		scanf("%d",&quant);
		if(quant==0){
			break;
		}
		int dX,dY;
		scanf("%d",&dX);
		scanf("%d",&dY);
		for(i=0;i<quant;i++){
			int x,y;
			scanf("%d",&x);
			scanf("%d",&y);
			if(x==dX||y==dY){
				printf("divisa\n");
			}
			else if(x>dX&&y>dY){
				printf("NE\n");
			}
			else if(x<dX&&y>dY){
				printf("NO\n");
			}
			else if(x<dX&&y<dY){
				printf("SO\n");
			}
			else if(x>dX&&y<dY){
				printf("SE\n");
			}
		}
	}
    return 0;
}
